src/zk.py

```diff
@@ -570,6 +570,13 @@ def get_alive_hosts(self, timeout=1, catch_except=True, all_hosts_timeout=None):
         if ha_hosts is None:
             return []
         if all_hosts_timeout:
-            timeout = max(timeout, all_hosts_timeout // len(ha_hosts))
+            minimal_total_timeout = timeout * len(ha_hosts)
+            if minimal_total_timeout > all_hosts_timeout:
+                logging.warning("The total timeout for checking the aliveness of all hosts (%s s) "
```

Comment:
Here we do not obey explicitly given `all_hosts_timeout`, we should also log that actually `minimal_total_timeout` will be used.
Warning only would be sufficient if we would use given `all_hosts_timeout`.